### PR TITLE
Support for div contentedittable in input directive added

### PR DESCRIPTION
**Pull request by abrararies**
I came across a problem where I needed to use a contenteditable div instead of input or textarea so I went ahead and made changes to the input directive. Take a look and let me know if its ok.

### CODE REVIEW

**Comment by abrararies** [comment]
Signed it.

**Comment by abrararies** [comment]
Signed it.

**Comment by abrararies** [comment]
Signed it.

**Comment by abrararies** [comment]
All the failed tests are under /src/components/select/select.js which I have not worked on.

**Comment by EladBezalel** [comment]
@abrararies please rebase and solve conflicts

**Comment by abrararies** [comment]
I've rebased the code, please check.

**Comment by EladBezalel** [comment]
I can't really understand why you would like that, before submitting this PR you should've opened an issue to discuss about this.

More than that please follow our [Commit Guidelines](https://github.com/angular/material/blob/master/docs/guides/CONTRIBUTING.md#-git-commit-guidelines) and make it 1 commit. thanks !

**Comment by abrararies** [comment]
So here is the story, for my recent project I needed to add images and icons to textarea. I needed to added text highlighting and fine manipulation to input and textarea elements.

Only way to achieve this is to use an editable div but such elements did not behave the same way as other input fields inside md-input-container, they lacked the animations, auto increment etc. Therefore, I forked and tweaked the code to support them.

My apologies for not discussing it with you. I just submitted the PR in hopes that it might help someone else. I hope that everything is in order now, let me know if you need anything else.

Thanks!

File: src/components/input/input.js [hunk 1]
```diff
@@ -312,7 +313,11 @@ function inputTextareaDirective($mdUtil, $window, $mdAria) {
     function inputCheckValue() {
       // An input's value counts if its length > 0,
       // or if the input's validity state says it has bad input (eg string in a number input)
-      containerCtrl.setHasValue(element.val().length > 0 || (element[0].validity || {}).badInput);
+      if(element[0].tagName.toLowerCase() !== 'div') {
```
Comment (DevVersion) [inline src/components/input/input.js]: Please space the actual if from the expression

**Comment by DevVersion** [comment]
Commit Message Style and Commit Amount looks good now!

**Comment by abrararies** [comment]
Done, please check.

**Comment by abrararies** [comment]
My bad. It wasn't done before, now it is. :)

**Comment by EladBezalel** [comment]
@abrararies are you talking about wysiwyg textarea? #6490

**Comment by abrararies** [comment]
This was intended to be just an addition to the types of input elements (input and textarea) that can be used inside md-input-container.
But yes, I guess anyone can use this to make a wysiwyg editor.

**Comment by ThomasBurleson** [comment]
This is not the best solution... better to create a new `contentEditable` directive.
Closing as too risky to to core functionality. 
Will not merge.

**Comment by yenoh2** [comment]
Here's my quick attempt at creating a new contentEditable directive.  There may be gaps in it but it'll give the look and feel of a textarea inside md-input-container.  It makes use of the same md-input-container controller like the textarea directive.  In fact it's the same directive with changes to 3 lines.  I posted it at the following location.  
https://groups.google.com/forum/#!topic/ngmaterial/vSIwNEsLK3k

**Pull request closed without merging**